h/static/scripts/annotation-mapper.js

```diff
@@ -18,11 +18,29 @@ function getContainer(threading, annotation) {
 
 // Wraps the annotation store to trigger events for the CRUD actions
 // @ngInject
-function annotationMapper($rootScope, threading, store) {
+function annotationMapper($rootScope, threading, store, groups) {
+
+  // All of the annotations that annotationMapper has received from calls to
+  // its loadAnnotations() method.
+  var received = [];
+
+  // The annotations that are currently loaded into the page context.
+  var loaded = [];
+
   function loadAnnotations(annotations) {
-    var loaded = [];
+    received = received.concat(annotations);
```

Comment:
This assumes that the same annotation never gets passed to `loadAnnotations()` twice, which may not be true. `received` may need to become a set, or a dict keyed by annotation id, or something,